File: Ordenação/Cpp/insertionSort.cpp
#include <iostream>

using namespace std;

int main(){

	int n, j, c, aux, aux2;

	cout << "Digite o Tamanho do vetor: ";
	cin >> n; // Pega do usuário um vetor de tamanho n.

	int vet[n]; // Inicializa o vetor com tamanho n.

	for(int z=0; z<n; z++){ // Alimentar o vetor com valores que o user digitar.
		cout << "Digite o " << z+1 << "º " << "valor do vetor: ";
		cin >> vet[z];
	}

	cout << "Antes do InserisonSort: " << endl;
	for(int a=0; a<n; a++){ // Mostrar vetor antes do InsertionSort
		cout << vet[a] << " ";
	}

	cout << endl;

	for(int i=1; i<n; i++){ // Início do InsertionSort Crescente

		j = i-1;
		aux = vet[i];

		while(j >= 0 && aux < vet[j]){

			vet[j+1] = vet[j];
			j--;
		}

		vet[j+1] = aux;

	}

	cout << "Depois do InserisonSort Crescente: " << endl;
	for(int dc=0; dc<n; dc++){ // Mostrar vetor depois do InsertionSort Crescente
		cout << vet[dc] << " ";
	}

	cout << endl;


	for(int l=1; l<n; l++){ // Início do InsertionSort Decrescente

		c = l-1;
		aux2 = vet[l];

		while(c >= 0 && aux2 > vet[c]){
			
			vet[c+1] = vet[c];
			c--;

		}

		vet[c+1] = aux2;

	}

	cout << "Depois do InserisonSort Decrescente: " << endl;
	for(int dd=0; dd<n; dd++){ // Mostrar vetor depois do InsertionSort Crescente
		cout << vet[dd] << " ";
	}

	cout << endl;

	return 0;

}

File: Ordenação/Python/insertionSort.py
def insertion(vt): # Função do InsertionSort, de parametro recebe um vetor.

	for i in range(1, len(vt)): # Para i no alcance de 0 até o tamanho do vetor.
		aux = vt[i] # Armazena na variavel auxiliar o valor do elemento do vetor na pos "i".
		j = i-1 # Valor de "j" depende da variável "i" - 1, ou seja, 1 posição antes de "i".

		while((j >= 0) and aux < vt[j]): # Enquanto "j" for maior ou igual a zero e var "aux"
										 # for menor que o valor do vet na pos "j" faça.
			vt[j+1] = vt[j] # O vet na pos "j"+1 recebe o vet na pos "j"
			j -= 1 # decrementa o "j" pra tentar sair do loop.
		
		vt[j+1] = aux # Por fim, o vet "j"+1 recebe o valor da var aux, que era o vet pos "i".

vet = [] # Inicializa a Lista/Vetor.
n = int(input("Digite o Tamanho do Vetor: ")) # "Restringe" o Tamanho de elementos.

for t in range(0,n): # Para t no alcance de 0 até o número inserido pelo user em "n".
	val = int(input("Digite o {}º elemento: ".format(t+1))) # Pega do usuario o valor pra add no vetor
	vet.append(val) # Adiciona o valor digitado no vetor

print("Vetor antes do InsertionSort: {}".format(vet))

insertion(vet) # Faz o InsertionSort

print("Vetor depois do InsertionSort: {}".format(vet))